File: solutions_5756407898963968_0/C++/Ideophage/A-MagicTrick.cpp
#include <cstdio>

void treatCase()
{
  bool poss[17];
  for (int i = 1; i <= 16; ++i)
    poss[i] = true;
  for (int l = 0; l < 2; ++l)
  {
    int j;
    scanf("%d", &j);
    for (int i = 1; i <= 4; ++i)
    {
      for (int r = 1; r <= 4; ++r)
      {
        int td;
        scanf("%d", &td);
        if (i != j)
          poss[td] = false;
      }
    }
  }
  int ok = -1;
  for (int i = 1; i <= 16; ++i)
    if (poss[i])
    {
      if (ok != -1)
      {
        printf("Bad magician!");
        return;
      }
      ok = i;
    }
  if (ok == -1)
    printf("Volunteer cheated!");
  else
    printf("%d", ok);
}

int main()
{
  int nbCases;
  scanf("%d", &nbCases);
  for (int iCase = 1; iCase <= nbCases; ++iCase)
  {
    printf("Case #%d: ", iCase);
    treatCase();
    printf("\n");
  }
  return 0;
}

File: solutions_5756407898963968_0/Python/MiaZ/A-magic_trick.py
n = int(raw_input())
for i in range(n):
	print "Case #%d:"%(i+1),
	
	fir =  int(raw_input())
	fir_mat = []
	for j in range(4):
		row = raw_input().split()
		for k in range(4):
			row[k] = int(row[k])
		fir_mat.append(row)
	 
	sec = int(raw_input())
	sec_mat = []
	for j in range(4):
		row = raw_input().split()
		for k in range(4):
			row[k] = int(row[k])
		sec_mat.append(row)	
		
	count = 0
	for j in range(4):
		flag = True
		for k in range(4):
			if sec_mat[sec-1][k]==fir_mat[fir-1][j]:
				count += 1
				ans = sec_mat[sec-1][k]
			if count > 1:
				ans = "Bad magician!"
				flag = False
				break
		if flag == False:
			break
	if count == 0:
		ans = "Volunteer cheated!"
		
	print ans